readme.md

```diff
@@ -23,3 +23,262 @@ For the revision task, please refer to: [revision.md](revision.md).
+      - There are three kinds: single precision(32 bits), double precision (64 bits) and extended double precision (80 bits).
+    ![alt text](image-24.png)
+
+- **Static, Extern, Volatile**
+  - **Static\:** 
+    - When applied to variables: It specifies that the variable has a local scope within a function or file scope (global/static variable) depending on where it's declared. If a variable is declared static within a function, its value persists between function calls.
+    <br>
+    - When applied to functions: It specifies that the function is only visible within the file it is declared in. It limits the scope of the function to the file it's defined in. 
+  - **Extern\:** 
+    - It is used to declare a variable or function that is defined in another file or in the main program. It allows users to use variables or functions across different files.
+    <br>
+    - When you declare a variable as extern, you're telling the compiler that the variable is defined elsewhere, and the linker will resolve the actual definition during linking.
+  - **Volatile\:**
+    - It tells the compiler that the variable can be changed unexpectedly. It prevents the compiler from performing certain optimizations on the variable.
+    <br>
+    - Typically, volatile is used for variables that can be changed by outside sources such as hardware, interrupts, or concurrently running threads.
+## 4. **Overflow**
+
+- **Definition\:**  phenomenon happens when users attempt to store inside a variable a value that is larger than the maximum value that the variable can hold.
+- **Example\:**
+  Integer oveflow with addition\:
+
+ ![alt text](image-19.png)
+ Integer overflow with multiplication\:
+
+ ![alt text](image-20.png)
+ Integer oveflow with conversion\:
+
+ ![alt text](image-21.png)
+ 
+ ![alt text](image-22.png)
+- **Ways to avoid overflow\:** 
+  - Add condition to check the overflow:
+  
+   ![Addition](image.png)
+   ![Subtraction](image-1.png)
+   ![Multiplication](image-2.png)
+   ![Division](image-3.png)
+   - Bounding check\: 
+  
+    ![alt text](image-23.png)
+  - Use available function of GCC compiler:
+  
+   ![alt text](image-4.png)
+## 5. **Format specifier**
+
+| **Types**          | **Format specifier** |
+| ------------------ | :--------------------: |
+| char               | `%c `                  |
+| unsigned char      | `%c`                   |
+| short              | `%hd `                 |
+| string             | `%s `                  |
+| int                | `%d`                   |
+| unsigned int       |` %u`                   |
+| long               | `%li `                 |
+| unsigned long      | `%lu`                  |
+| long long          | `%lli`                 |
+| unsigned long long | `%llu `                |
+| float              | `%f `                  |
+| double             | `%lld`                 |
+| long double        | `%lf`                  |
+
+- **Extended format specifier\:**
```

Comment:
@spm1102 
`#reply_required`
`#fix_required`

There are still some formats (Hint: Explain the behaviour of this format specifier: %2s, %02d)